src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -3106,6 +3113,58 @@ bool KeyCondition::alwaysFalse() const
     return rpn_stack[0] == 0;
 }
 
+bool KeyCondition::canBeFalseAlwaysUnknown() const
```

Comment:
I don't fully understand the implementation.
If we have the function `canBeTrueAlwaysUnknown`, what would be the difference?

Reply:
> If we have the function canBeTrueAlwaysUnknown, what would be the difference?

The main difference would be the implementation of `RPNElement::FUNCTION_OR` and `RPNElement::FUNCTION_AND`.

`atom_can_be_false_always_unknown AND any_atom   => atom_can_be_false_always_unknown`

`atom_can_be_true_always_unknown OR any_atom => atom_can_be_true_always_unknown`

`canBeTrueAlwaysUnknown` is introduced as an optimization to find exact ranges during key analysis. If the given key condition is not possible to tell some hyperrectangle can be false, exact ranges cannot be determined. We'll disable exact range finding and fall back to faster analysis by checking `.can_be_true` only.

 in a special case.